### PR TITLE
Fix and enhance `scripts/update-distribution-scripts.sh`

### PR DESCRIPTION
**Pull request by straight-shoota**
The script was not working correctly, the generation of commit messages was broken.

This patch also makes the branch configurable and creating a PR optional (this is useful when creating a branch for a test run).

### CODE REVIEW

**Review by beta-ziliani — APPROVED** [review]
I trust the sed change, the rest is 👍

File: scripts/update-distribution-scripts.sh [hunk 1]
```diff
@@ -38,15 +42,19 @@ sed -i -E "/distribution-scripts-version:/{n;n;n;s/default: \".*\"/default: \"$r
 git add .circleci/config.yml
 
 message="Updates \`distribution-scripts\` dependency to https://github.com/crystal-lang/distribution-scripts/commit/$reference"
-log=$($GIT_DS log $old_reference..$reference --format="%s" | sed "s/.*(/\* crystal-lang\/distribution-scripts/;s/.$//")
-message="$message\n\nThis includes the following changes:\n\n$log"
+log=$($GIT_DS log $old_reference..$reference --format="%s" | sed "s/.*(/crystal-lang\/distribution-scripts/;s/^/* /;s/.$//")
+message=$(printf "$message\n\nThis includes the following changes:\n\n$log")
```
Comment (HertzDevil) [inline scripts/update-distribution-scripts.sh:46]: Will this break if the message contains `%`?

Comment (oprypin) [reply scripts/update-distribution-scripts.sh:46]: Yes. The answer is yes 😅
```suggestion
message=$(printf "%s" "$message\n\nThis includes the following changes:\n\n$log")
```

Comment (oprypin) [reply scripts/update-distribution-scripts.sh:46]:
```suggestion
message=$(printf "%s\n\nThis includes the following changes:\n\n%s" "$message" "$log")
```

**Pull request merged**